### PR TITLE
[cxx] Do not define C++ keywords: bool, inline, final, it can break things.

### PR DESCRIPTION
**Pull request by jaykrell**
https://jenkins.mono-project.com/job/d/421/parsed_console/log.html

  C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\xkeycheck.h(250): fatal error C1189: #error:  The C++ Standard Library forbids macroizing keywords. Enable warning C4005 to find the forbidden macro. [d:\j\workspace\d\msvc\libmonoruntime.vcxproj]

### CODE REVIEW

**Comment by jaykrell** [comment]
@monojenkins build failed

File: mono/eglib/eglib-config.hw [hunk 1]
```diff
@@ -45,8 +45,10 @@ typedef int pid_t;
 #define F_SETFD 1
 #define FD_CLOEXEC 1
 
+#ifndef __cplusplus
 #undef inline
 #define inline __inline
+#endif
```
Comment (jaykrell) [inline mono/eglib/eglib-config.hw]: This is probably the crucial one.

**Comment by jaykrell** [comment]
@monojenkins build failed

File: libgc/include/private/gc_locks.h [hunk 2]
```diff
@@ -337,7 +337,9 @@ extern "C" {
          }
 #      else
 #        ifndef inline
-#          define inline __inline
+#          ifndef __cplusplus
+#            define inline __inline
```
[the remark this answers is not in the document]
Comment (jaykrell) [reply libgc/include/private/gc_locks.h]: Yep, they fixed it in 2011.
https://github.com/ivmai/bdwgc/commit/e35a4171fe47dfbf847e08988ea6cec4dfc8d124#diff-08bcd882fb4fadbb0c77a0926ef80484L315

**Comment by akoeplinger** [comment]
@monojenkins backport 2018-10

**Pull request merged**